### ISSUE TITLE
Angular 2 test cases

### ISSUE THREAD

**Issue by deepu105** [jhipster/generator-jhipster#4301]
This is a tracker in order to test Angular 2 migration @jhipster/hipsters please add on If I missed anything
Related to #3664 

**Application**
- [x] Default options with i18n
- [x] Default options without i18n
- [x] Default options with i18n prod profile
- [x] Default options without i18n prod profile
- [x] Default options regenerate existing app
- [x] Default options with SASS
- [ ] Default options with Social Login
- [x] Default options with `--skip-user-management` flag
- [x] Default options with `--jhi-prefix` flag
- [x] Default options with `--skip-server` flag
- [x] Default options with i18n for Gateway
- [x] Default options without i18n for Gateway
- [ ] Default options with i18n for UAA server
- [ ] Default options without i18n for UAA server
- [x] Default options with Protractor
- [x] Gradle build Default with i18n
- [x] Gradle build Default without i18n
- [x] Gradle build Default with i18n prod profile
- [x] Gradle build Default without i18n prod profile
- [x] With Search engine
- [x] With websocket
- [x] JWT Auth
- [ ] Oauth2 Auth
- [ ] add new languages to app

**Entity**
- [x] With no pagination
- [ ] With pager
- [x] With pagination
- [x] With Infinite scroll
- [ ] With blob
- [ ] With Image
- [x] With Text
- [ ] Different field types
- [ ] Different relationship types
- [x] Import from JDL

**Comment by VictorADS** [comment]
Options I tested and that worked :

- [x] Default options with i18n
- [x] Default options without i18n
- [x] Default options regenerate existing app ( I had entities and i needed to regenerate them all to make it work)
- [x] With no pagination
- [x] With pager
- [x] With pagination
- [x]  With Text
- [x] Different relationship types
- [x] Import from JDL

Needs more testing : 
- [ ] Different field types (Not sure  EnumField generation works well)

Didnt work : 
- [ ] With blob
- [ ] With Image

**Comment by xetys** [comment]
I am currently testing the default options with SASS enabled. I know SASS is not our main focus, but it is not really working, as:

* hot reload is working somekind strange, not for all files
* you can change the main.scss, but it doesn't affect the main.css

so 

didn't work

 - [ ] Default with SASS

**Comment by jdubois** [comment]
You also need to test entities with Elasticsearch, with and without i18n at least.

**Comment by deepu105** [comment]
Test cases reset as we switched to Ng2 router :(

**Comment by pascalgrimaud** [comment]
Some issues I found this morning:
- maven build in dev:
    - invoke the run webpack:build:dev : discussion with @deepu105 here https://github.com/jhipster/generator-jhipster/commit/88ba738f406b656ae1e9dd68517dac33929a290c#commitcomment-20570911

- pagination
    - in user management page, the pagination doesn't work
    - when generate an ng2 project (with BankAccount, Operation and Label as entities), the Label page can't be displayed, because of that code: [?page&sort&search](https://github.com/jhipster/generator-jhipster/blob/master/generators/entity/templates/client/angular/src/main/webapp/app/entities/_entity-management.state.ts#L47)
    - so I did the same in [user-management page](https://github.com/jhipster/generator-jhipster/blob/master/generators/client/templates/angular/src/main/webapp/app/admin/user-management/_user-management.state.ts#L43) and I delete the `?page&sort&search`, the page is displayed correctly (as in user management), but the pagination won't work

**Comment by deepu105** [comment]
@jhipster/developers guys moe hands here please. Plz tick off the options that works without any issues (except known issues if any) If there are issues and if you cant do PR then plz open a ticket so that me or someone else can try to fix it

**Comment by pascalgrimaud** [comment]
Here all Travis builds with only ng2 app:
https://travis-ci.org/pascalgrimaud/generator-jhipster/builds/195433340

It seems to be an issue with elasticsearch which should be fixed in this PR https://github.com/jhipster/generator-jhipster/pull/4970

It wont replace manually tests but I think it can help

**Comment by deepu105** [comment]
Can you also try to enable protractor and Karma tests on all options?

Thanks & regards,
Deepu

**Comment by pascalgrimaud** [comment]
Karma tests are already enabled for all options.
And for protractor, yes I can! I will do that

**Comment by pascalgrimaud** [comment]
With protractor enabled for all builds: https://travis-ci.org/pascalgrimaud/generator-jhipster/builds/195487021

- Cassandra + ng2: fail on protractor tests
- OAuth2 + ng2: fail on protractor tests

**Comment by deepu105** [comment]
Thanks @pascalgrimaud I was wondering if we could test most of the options here with some kind of automation. with some script or something that can iterate through options and generate apps according to the options and build, lint, test. we cover some cases in travis but it might be nice to have such a script so that we can test everything locally (or in docker or something)

**Comment by pascalgrimaud** [comment]
@deepu105 : what do you think about using the Travis builds from hipster labs orga ? As it is free and not used as often as jhipster orga

**Comment by deepu105** [comment]
Yes we can use that if thats possible

**Comment by deepu105** [comment]
also how difficult it is to configure Travis to run protractor tests on all builds when something is merged to master but not for PR builds?

**Comment by pascalgrimaud** [comment]
Ok I ll give a try. Maybe I will need you to connect the repo to Travis, so I will ping you :P

**Comment by pascalgrimaud** [comment]
Not difficult, just a variable to detect the branch which is currently tested by Travis.
So we can have a better build when merging into master, and a lighter build for PR. Or the other side if you prefer

**Comment by deepu105** [comment]
@pascalgrimaud yes it would be nice not to tax travis too much so I guess we could add protractor to alls config only when the branch is master what do you think @jdubois 

Btw im fixing the 2 broken builds in your branch on our master

**Comment by deepu105** [comment]
@jdubois @pascalgrimaud since we have frozen ng1 development it should stay stable hence what about switching some of the travis builds to test ng2 and keep ng1 only for few something like below

Ng1:
```
    - JHIPSTER=app-default-from-scratch PROFILE=prod RUN_APP=0
    - JHIPSTER=app-gradle PROTRACTOR=1
    - JHIPSTER=app-psql-es-noi18n PROFILE=prod PROTRACTOR=1
    - JHIPSTER=app-mysql PROFILE=prod PROTRACTOR=1
    - JHIPSTER=app-jwt
    - JHIPSTER=app-hazelcast-cucumber
    - JHIPSTER=app-mariadb-kafka
    - JHIPSTER=app-microservice-eureka
    - JHIPSTER=app-microservice-consul
```
Ng2:
```
    - JHIPSTER=app-social-websocket PROTRACTOR=1
    - JHIPSTER=app-gateway-eureka
    - JHIPSTER=app-gateway-consul
    - JHIPSTER=app-gateway-uaa
    - JHIPSTER=app-cassandra
    - JHIPSTER=app-mongodb
    - JHIPSTER=app-oauth2
    - JHIPSTER=app-ng2-default PROTRACTOR=1
    - JHIPSTER=app-ng2-jwt PROTRACTOR=1
    - JHIPSTER=app-ng2-sass-noi18n PROTRACTOR=1
```

**Comment by pascalgrimaud** [comment]
I woud prefer switching full ng2, and keeping only 3 builds for ng1

Travis provides cron jobs: https://docs.travis-ci.com/user/cron-jobs/ (I never test it!!)
I can create a repo on Hipster labs, and every nights, the same builds can run for NG1, so there are no regression
I can create a full build for gradle too :)
There are a lot of possibilities, simply discuss and tell what you want

**Comment by deepu105** [comment]
I kept the other builds in ng1 as `hazelcast-cucumber, and `mariadb-kafka` really shouldnt depend on clientSIde so we could double them as ng1 test as well and 2 of them are microservice so no cleint, `jwt` is already on ng2 so the other one can stay for ng1. mysql and plsql shouldnt differ based on client and our ng2 prod is not proper yet so lets keep it for ng1 now we can later switch them.

**Comment by deepu105** [comment]
@pascalgrimaud you are the travis expert here so I'll follow your lead for setting stuff up in labs org.

**Comment by deepu105** [comment]
Let me know once you create the repo there i'll make you its admin so that you can setup stuff

**Comment by pascalgrimaud** [comment]
@deepu105 :
- Just created the repo here: https://github.com/hipster-labs/jhipster-travis-build (hope you're ok with the name :p)
- The 1st travis build in Hipster labs organization: https://travis-ci.org/hipster-labs/jhipster-travis-build/builds/195652235

**Comment by deepu105** [comment]
another way to decrease build time and reduce loading travis is to use one more free CI to run some builds  in parallel :) may be even AppVoyer or circleCI?

**Comment by pascalgrimaud** [comment]
Sure it's possible with CircleCI. For AppVeyor, I never use it as I dont know powershell syntax :p

About https://github.com/hipster-labs/jhipster-travis-build, now we have same (current/old) builds here: https://travis-ci.org/hipster-labs/jhipster-travis-build/
I param a CRON to trigger automatically a build every 24h <- I need to confirm it works
Or you can launch a build by pushing text in a dummy.txt file

We can param a webhook but it should be too much... every 24h should be ok

**Comment by deepu105** [comment]
On the build console log it says its forking that project. How are we telling it to fork this project and run tests on it?

**Comment by deepu105** [comment]
Oh sorry I see that you are doing that from a script

**Comment by pascalgrimaud** [comment]
I will check everything tomorrow morning, time to sleep :sleeping:

**Comment by jdubois** [comment]
- yes let's go NG 2 with our "normal" Travis builds, as NG 1 is frozen.
- I'm curious to see what it does with our Sonar analysis
- for NG 1 we could have a daily batch working with "hipster labs", that's good enough.
- I'd rather have everything on Travis than use different systems, it's already too complex like this
- I'll contact Travis to see if we can have something better. And if we need to pay, I'll look for a "sponsor".

**Comment by pascalgrimaud** [comment]
@jdubois : I didn't change the project `app-default-from-scratch` -> jhipsterSampleApplication which is concerned by the Sonar analysis. It is still with NG1
Do you want to switch to NG2 ?

**Comment by jdubois** [comment]
Yes let's try doing the Sonar analysis on NG 2. No idea if Sonar
understands TypeScript !

Le 27 janv. 2017 7:55 AM, "Pascal Grimaud" <notifications@github.com> a
écrit :

**Comment by VictorADS** [comment]
@deepu105 I'm Currently testing the `--skip-server` flag and it appears to be broken on both ng1 and ng2.
It breaks because of a` Depency Injection Error ` since we have no `AuthServerProvider`. It is easily fixable but I would like to know what is the use case behind this option ?

**Comment by deepu105** [comment]
It is for people who want to generate a front end app seperateley. There are cases where you would like to have your client and server deployed in different containers. It should be easily fixable

**Comment by deepu105** [comment]
@jdubois I dont think it will become too complex. Actually I think it would be quite the opposite. Its just about writing another `circle.yml` and then we can then split our builds into 2 parallel builds thus reducing build time and also increasing coverage as we can then enable Protractor on all builds without worrying about build time increase. It would be much easier than finding a sponsor I believe

**Comment by wmarques** [comment]
I checked some options with @VictorADS

**Comment by jdubois** [comment]
- Social login doesn't work, both with AngularJS 1 and Angular 2
  - With Angular 2, the login looks to work, but there is a stacktrace showing a registration email being sent (which shouldn't happen as we use Social login), and then we get redirected to a page with no view (there's the menu and footer, but nothing in the middle)
  - With AngularJS 1 I can't get authenticated by Google (and I'm using the same codes as for Angular 2)
- On all projects, when I do a "prod" build, I still have a banner (the "ribbon" that shows we are in dev mode), with a message "no translation"
- With JWT, I can't test with Swagger: when I click on "try it out" to test an API, it is not authenticated as the JWT token isn't sent. This needs to be checked also for OAuth2

**Comment by jdubois** [comment]
- I can't login with UAA. Everything starts up correctly, but the login form gets a 404 error. @xetys if you have time to check...

**Comment by jdubois** [comment]
- Websocket is checked as it works in "prod" mode, but in "dev" mode there is no way to use it correctly: with just HMR (port 9060) or with BrowserSync + HMR (port 9000) the socket can't be opened, so it's impossible to develop easily.

**Comment by ruddell** [comment]
@jdubois I didn't have any issues with NG1 and Google, is there a specific error message?  Also we send an email on account creation for social users

For NG2, social-auth.register.ts doesn't get the `provider`/`success` params from the route.  I don't think ` this.provider = params['provider{success:boolean}'];` looks right so I'll leave that to someone who knows ngRouter.  For easier testing, the URL that breaks is http://localhost:9000/#/social-register/twitter?success=true

**Comment by jdubois** [comment]
@ruddell for Social login + NG1, I get:

    The redirect URI in the request, http://localhost:8080/signin/google, does not match the ones authorized for the OAuth client

But this URL is correctly configured, and works with NG2 (with NG2 it fails after that, when displaying the page)

**Comment by jdubois** [comment]
And the last bullet point "add new languages to app" doesn't work...

-> I'm going to finish testing all those, and then I'll close this ticket and create one specific ticket per issue, as there are just too many of them here

**Comment by jdubois** [comment]
Testing the entities:

- Dates cannot be edited (both LocalDate and ZoneDateTime)
- ZoneDateTime do not have the correct pop up, as the time can't be set up
- The "pager" and "pagination" are the same component
- In many-to-many relationships, when you view the entity, you cannot click on the "many" entities

**Comment by deepu105** [comment]
@jdubois I see the Oauth option is checked but what about https://github.com/jhipster/generator-jhipster/issues/4982 can someone take a look at it? I'm not an Oauth expert

**Comment by jdubois** [comment]
@deepu105 oh I didn't test this, so OK let's uncheck this.
Maybe we should remove the OAuth2 option, it doesn't make much sense (you'd better have an external OAuth2 server, for what we do here JWT is much better and probably more secure), causes lots of issues, and nearly nobody uses it (only 13 generations last month!!). I'll do a ticket for this too.

**Comment by jdubois** [comment]
I've tested everything, I'm going to:

- Create tickets for each issue
- Give them priorities

If we can't solve them easily, we can still remove options in the generators.

And I'll close this ticket afterwards as it is becoming too big.

**Comment by deepu105** [comment]
@jdubois IMO lets not remove options as we are only releasing a Beta of angular support and not final. if we remove options we wont get feedback for those as failing options would prompt more contribution from people who try to use them and we can fix them once we move NG2 support out of Beta

**Comment by jdubois** [comment]
@deepu105 it's not really cool to allow options that we know are failing. We should at the very least put a clear warning, like "unfinished".

**Comment by jdubois** [comment]
@deepu105 I'm listing all those errors, I still have good hope we can solve them all.

**Comment by deepu105** [comment]
Agree but isnt marking something as [BETA] clear warning that its unfinished? 
Else we can mark such options [Unstable] so that people still get a chance to use them and fix them if its required for them. if we dont give people a way to use them at all then the burden of fixing those options will be solely on us.

**Comment by jdubois** [comment]
Closing as I added all the issues as specific tickets!

**Comment by deepu105** [comment]
Did anyone test out an option with a custom jhiPrefix added?


Thanks & Regards,
Deepu

**Comment by gmarziou** [comment]
Just tried and it does not work.

~~~
Template parse errors:↵Can't bind to 'predicate' s…iTranslate="global.fie"): UserMgmtComponent@11:78"} Error: Template parse errors:
Can't bind to 'predicate' since it isn't a known property of 'tr'. ("
        <table class="table table-striped">
            <thead>
            <tr exampleSort [ERROR ->][(predicate)]="predicate" [(ascending)]="reverse" [callback]="transition.bind(this)">
            <th"): UserMgmtComponent@11:28
Can't bind to 'ascending' since it isn't a known property of 'tr'. ("ss="table table-striped">
            <thead>
            <tr exampleSort [(predicate)]="predicate" [ERROR ->][(ascending)]="reverse" [callback]="transition.bind(this)">
            <th exampleSortBy="id"><span "): UserMgmtComponent@11:54
Can't bind to 'callback' since it isn't a known property of 'tr'. (">
            <thead>
            <tr exampleSort [(predicate)]="predicate" [(ascending)]="reverse" [ERROR ->][callback]="transition.bind(this)">
            <th exampleSortBy="id"><span jhiTranslate="global.fie"): UserMgmtComponent@11:78
~~~

**Comment by deepu105** [comment]
Ya I tried as well and fixed most of the issues
